From 7f5125e81f37d8904b7fcfd4075b87fc883acd94 Mon Sep 17 00:00:00 2001
From: "p.bred" <p.bred@samsung.com>
Date: Thu, 11 Aug 2022 04:12:42 +0300
Subject: [PATCH 029/161] [drivers/dax/axdimm] Add module version (#11)

* the version can be obtained: modinfo -Fversion device_dax
* append dirty mark if there are local changes

Signed-off-by: Petr Bred <p.bred@samsung.com>
---
 drivers/dax/Makefile     | 1 +
 drivers/dax/axdimm.c     | 4 ++++
 scripts/package/builddeb | 1 -
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/dax/Makefile b/drivers/dax/Makefile
index f15a4775b..7af067884 100644
--- a/drivers/dax/Makefile
+++ b/drivers/dax/Makefile
@@ -7,6 +7,7 @@ obj-$(CONFIG_DEV_DAX_PMEM) += dax_pmem.o
 dax-y := super.o
 dax-y += bus.o
 device_dax-y := device.o
+CFLAGS_axdimm.o += -DAXDIMM_DRIVER_VERSION=\"$(shell git describe --first-parent --abbrev=0 --dirty --always)\"
 device_dax-y += axdimm.o
 device_dax-y += axdimm_allocator.o
 dax_pmem-y := pmem.o
diff --git a/drivers/dax/axdimm.c b/drivers/dax/axdimm.c
index 3c4c1a8e6..7b6b4f9fe 100644
--- a/drivers/dax/axdimm.c
+++ b/drivers/dax/axdimm.c
@@ -17,6 +17,10 @@
 #include <linux/mman.h>
 #include <linux/io.h>
 
+#ifdef AXDIMM_DRIVER_VERSION
+MODULE_VERSION(AXDIMM_DRIVER_VERSION);
+#endif
+
 #define AXDIMM_SEM_VALUE 1
 
 typedef struct axdmem_info axdmem_info;
diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index aa599a578..67cd420dc 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -89,7 +89,6 @@ deploy_kernel_headers () {
 
 	mkdir -p $pdir/lib/modules/$version/
 	ln -s /usr/src/linux-headers-$version $pdir/lib/modules/$version/build
-	git log -n 1 --pretty=format:"%H%n%n%B%n" > $pdir/lib/modules/$version/axdimm_info
 }
 
 deploy_libc_headers () {
-- 
2.34.1

